src/platforms/pipedrive/schemas.ts

```diff
@@ -0,0 +1,234 @@
+      })
+      .optional(),
+    additional_data: z
+      .object({
+        properties: z
+          .object({
+            more_items_in_collection: z.boolean().optional(),
+            start: z.number().optional(),
+          })
+          .optional(),
+      })
+      .optional(),
+  });
+
+export const pipedriveModuleSchema = z.enum(PIPEDRIVE_MODULES);
+export type PipedriveModule = (typeof PIPEDRIVE_MODULES)[number];
+
+export const upsertResponseSchema = z.object({
+  id: z.string(),
+});
+export type UpsertResponse = {
+  id: string;
+};
+
+// -
+// Subtypes
+// -
+export type PipedriveEmailAddressType = (typeof EMAIL_TYPES)[number];
+export const pipedriveEmailAddressSchema = z.object({
+  primary: z.boolean(),
+  label: z.string().transform((l) => l as PipedriveEmailAddressType),
+  value: z.string(),
+});
+
+export type PipedrivePhoneNumberType = (typeof PHONE_TYPES)[number];
+export const pipedrivePhoneNumberSchema = z.object({
+  primary: z.boolean(),
+  label: z.string().transform((l) => l as PipedrivePhoneNumberType),
+  value: z.string(),
+});
+
+export const pipedriveAssociationId = z.object({
+  value: z.number(),
+  name: z.string(),
+});
+
+// -
+// Users
+// -
+export const pipedriveUserSchema = z.object({
+  id: z.number(),
+  name: z.string(),
+  email: z.string(),
+  created: custom.date(),
+  modified: custom.date().optional(),
+  active_flag: z.boolean(),
+});
+export type PipedriveUser = z.infer<typeof pipedriveUserSchema>;
+
+// -
+// Person
+// -
+const pipedrivePersonSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  first_name: z.string(),
+  last_name: z.string(),
+  email: z.array(pipedriveEmailAddressSchema),
+  phone: z.array(pipedrivePhoneNumberSchema),
+  add_time: custom.date(),
+  update_time: custom.date(),
+  active_flag: z.boolean(),
+  org_id: pipedriveAssociationId.optional(),
+  owner_id: pipedriveAssociationId.optional(),
+});
+export type PipedrivePerson = z.infer<typeof pipedrivePersonSchema>;
+
+export const pipedrivePersonUpsertSchema = z
+  .object({
+    name: z.string(),
+    first_name: z.string(),
+    last_name: z.string(),
+    email: z.array(pipedriveEmailAddressSchema),
+    phone: z.array(pipedrivePhoneNumberSchema),
+    org_id: z.number().optional(),
+    owner_id: z.number().optional(),
+    $native: z.record(z.any()),
+  })
+  .partial();
+export type PipedrivePersonCreate = z.infer<typeof pipedrivePersonUpsertSchema>;
+export type PipedrivePersonUpdate = z.infer<
+  typeof pipedrivePersonUpsertSchema
+> & { id: string };
+
+// -
+// Deals
+// -
+export const pipedriveDealSchema = z.object({
+  id: z.number(),
+  title: z.string(),
+  stage_id: z.number(),
+  value: z.string(),
+  expected_close_date: custom.date(),
+  weighted_value: z.string(),
+  status: z.string(),
+  active: z.boolean(),
+  probability: z.string(),
+  update_time: custom.date(),
+  add_time: custom.date(),
+  deleted: z.boolean(),
+  person_id: z.object({ value: z.number() }),
```

Comment:
do these need `.passthrough`?

Reply:
We decided on removing .passthrough for the client schemas (though I do think we should bring them back for retrieving $native fields)